### DISCUSSION TITLE
多列合并行时，表格数据缺失。

### DISCUSSION THREAD

**Discussion by lushdog** [ant-design/ant-design#35412] [General]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/biao-ge-xing-lie-he-bing-antd-4-20-1-forked-sef5lq?file=/demo.js)

### Steps to reproduce

```javascript
[
  {"key":"0","category":"水果","name":"桃子","desc":"好吃","categoryrowSpan":2,"namerowSpan":1,"descrowSpan":3},
  {"key":"1","category":"水果","name":"茄子","desc":"好吃","categoryrowSpan":0,"namerowSpan":1,"descrowSpan":0},
  {"key":"3","category":"家禽","name":"牛肉","desc":"好吃","categoryrowSpan":3,"namerowSpan":2,"descrowSpan":0},
  {"key":"4","category":"家禽","name":"牛肉","desc":"好吃到停不下来","categoryrowSpan":0,"namerowSpan":0,"descrowSpan":1},
  {"key":"5","category":"家禽","name":"猪肉","desc":"吃不起，太贵","categoryrowSpan":0}
] 
```
五条数据经过动态计算rowSpan后，表格只有4条。看rowSpan的数据是正确的。

### What is expected?

正确显示

### What is actually happening?

少了一条数据
"家禽","name":"牛肉","desc":"好吃"

| Environment | Info |
| --- | --- |
| antd | 4.20.1 |
| React | 17-18 |
| System | Mac/Window |
| Browser | Chrome |

**Comment by afc163** [comment]
<img width="233" alt="图片" src="https://user-images.githubusercontent.com/507615/167235407-62a4b1b7-50cf-404a-b9dd-fbc6c3b10848.png">

第四条数据的 rowSpan 为 0，位置被第三条（rowSpan 为 2）占据了。

**Comment by lushdog** [comment]
> <img alt="图片" width="233" src="https://user-images.githubusercontent.com/507615/167235407-62a4b1b7-50cf-404a-b9dd-fbc6c3b10848.png">
> 
> 第四条数据的 rowSpan 为 0，位置被第三条（rowSpan 为 2）占据了。

第三条和第四条只有desc不一样，categroy和name就应该被第三条合并。

**Comment by afc163** [comment]
key 3 这一条，descrowSpan 应该为 1，这里应该是算错了。

这个数据结构本身对应的展现是符合组件预期的。

**Comment by lushdog** [comment]
> key 3 这一条，descrowSpan 应该为 1，这里应该是算错了。
> 
> 这个数据结构本身对应的展现是符合组件预期的。

key 3 这一条，descrowSpan 设置为 1，最后边会突出来。
发现问题是一条数据没有rowSpan为1的，就不能显示。现在只能多加一个无用的列，然后隐藏。
![图标](https://static.mcc.miui.com/cms/a37752d7ec08e006/1/1651893968397.png)
![图标](https://static.mcc.miui.com/cms/a37752d7ec08e006/1/1651894314927.png)

**Comment by afc163** [comment]
> key 3 这一条，descrowSpan 设置为 1，最后边会突出来。

第一项的 descrowSpan 应该为 2 。
